src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs

```diff
@@ -688,7 +688,7 @@ public async Task WebHostStopAsyncUsesDefaultTimeoutIfNoTokenProvided()
         }
 
         [Fact]
-        public async Task HostPropagatesExceptionsThrownWithBackgroundServiceExceptionBehaviorOfStopHost()
+        public async Task HostStopsWithBackgroundServiceExceptionBehaviorOfStopHost()
```

Comment:
So this wasn't guaranteed before, it was only happening because the test passed `Task.CompletedTask` so the exception happened synchronously in the call to `ExecuteAsync` - no propagation was happening, it was just normal exception behavior.

There's nothing here that actually is meant to propagate an exception to a different thread, just stop the service:https://github.com/dotnet/runtime/blob/05e5a448fe726617ed881620093a792f27371e71/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L187-L204

So I've adjusted this test case to just ensure that the lifetime stopping is triggered.